_shared/local_extensions.py

```diff
@@ -158,6 +160,60 @@ def dedent(self, s):
         """Return `s` dedented."""
         return dedent(s)
 
+    def random_port_number(self):
+        """Return a randomly-generated port number."""
+        if not hasattr(self, "_random_port_numbers"):
+            self._random_port_numbers = [
+                i
+                for i in range(1025, 10000)  # Ports <=1024 are privileged on Linux.
+                if i
+                not in (
+                    # Commonly-used port numbers to avoid.
+                    8080,  # Common alternative HTTP port.
+                    # Port numbers already used by existing Hypothesis apps.
+                    5000,  # h's web process.
```

Comment:
The code formatting is Black's, not mine :man_shrugging: Including the lack of empty lines between the different sections of the list.

Reply:
Yeah I don't like this, as I often try to visually group things and then `black` smashes them all together.

You can space them out with comments, but it's still a bit ugly.